src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClientBase.h

```diff
@@ -65,6 +67,33 @@ namespace client
     /* Non-template base client class that contains main Aws Client Request pipeline logic */
     class SMITHY_API AwsSmithyClientBase
     {
+    private:
+        const char* ALLOC_TAG{"AwsSmithyClientBase"};
+
+
+        void deepCopyClientConfiguration(const AwsSmithyClientBase& target)
+        {
+            //first create copy from target
+            m_clientConfig = Aws::MakeShared<Aws::Client::ClientConfiguration>(target.m_serviceName.c_str(), *target.m_clientConfig);
```

Comment:
please make `Aws::MakeShared` with `m_serviceName` and not target. they should be the same, but lets be consistent.